src/compiler/checker.ts

```diff
@@ -9757,7 +9757,12 @@ module ts {
         function checkClassDeclaration(node: ClassDeclaration) {
             // Grammar checking
             if (node.parent.kind !== SyntaxKind.ModuleBlock && node.parent.kind !== SyntaxKind.SourceFile) {
-                grammarErrorOnNode(node, Diagnostics.class_declarations_are_only_supported_directly_inside_a_module_or_as_a_top_level_declaration);
+                grammarErrorOnFirstToken(node, Diagnostics.class_declarations_are_only_supported_directly_inside_a_module_or_as_a_top_level_declaration);
+            }
+
+            // node.flags & NodeFlags.Default || kind === SyntaxKind.ClassExpression

```

Comment:
Turn this into a TODO.

Also, for future reference, I am more in favor of `ClassyDeclaration` over `ClassLikeDeclaration`.